examples/gtest_emulation/CMakeLists.txt

```diff
@@ -0,0 +1,38 @@
+# Copyright (c) 2020  ETH Zurich
+#
+# SPDX-License-Identifier: BSL-1.0
+# Distributed under the Boost Software License, Version 1.0. (See accompanying
+# file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+
+cmake_minimum_required(VERSION 3.13)
+
+project(gtest CXX)
+
+if(EXISTS "${HPX_DIR}")
+
+  find_package(HPX REQUIRED)
+
+  # Add a static library which contains a main to emulate gtest_main
+  add_library(static_main_lib STATIC static_main.cpp)
+
+  # /!\ This helper interface is needed to keep the right linking order
+  add_library(hpx_helper_interface INTERFACE)
```

Comment:
So was this indirection through an interface library needed in the end or not?

Reply:
needed otherwise the link order is not the right one, cmake is weird on that point